### PR TITLE
RT-15627: add Subscriptions endpoints to apiV2

### PR DESCRIPTION
**Pull request by tumbochka**

### CODE REVIEW

File: src/Account/Subscriptions/CancelSubscription.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { APICall, APIFetchSettings } from "../../types";
+import { Subscription, SubscriptionCancel } from "./types";
+import getBaseURL from "../../utils/getBaseURL";
+import getFetchConfiguration from "../../utils/getFetchConfiguration";
+
+export default CancelSubscription as APICall<SubscriptionCancel, Subscription>;
+
+async function CancelSubscription(settings: APIFetchSettings, body: SubscriptionCancel) {
+  const { id } = body;
```
Comment (jeffda-self) [inline src/Account/Subscriptions/CancelSubscription.ts:9]: I would remove this line (#9) and destructure the argument on line #8.
```
async function CancelSubscription(settings: APIFetchSettings, { id }: SubscriptionCancel) {
    ...
}
```

File: src/Account/Subscriptions/CancelSubscription.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  const url_path = `/api/account/subscriptions/${id}`;
+  const base_url = getBaseURL(settings);
+  const fetch_configuration = getFetchConfiguration({ method: "DELETE", ...settings });
```
Comment (jeffda-self) [inline src/Account/Subscriptions/CancelSubscription.ts:12]: Do we want to force method "DELETE" or allow the settings argument to override it?

Comment (tumbochka) [reply src/Account/Subscriptions/CancelSubscription.ts:12]: We want to force `DELETE`.

File: src/Account/Subscriptions/GetSubscriptions.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { APICall, APIFetchSettings } from "../../types";
+import { Subscription, SubscriptionQuery } from "./types";
+import getBaseURL from "../../utils/getBaseURL";
+import toQueryString, { QueryObject } from "../../utils/toQueryString";
+import getFetchConfiguration from "../../utils/getFetchConfiguration";
+
+export default GetSubscriptions as APICall<SubscriptionQuery, Array<Subscription>>;
+
+async function GetSubscriptions(settings: APIFetchSettings, query: SubscriptionQuery | null) {
```
Comment (jeffda-self) [inline src/Account/Subscriptions/GetSubscriptions.ts:9]: Should the optional type for the query parameter be null or void? Will it be explicitly set to "null" or could it be omitted entirely thus passing "undefined"?

**Review by jeffda-self — CHANGES_REQUESTED** [review]

File: src/Account/Subscriptions/GetSubscriptions.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import type { APICall, APIFetchSettings } from "../../types";
+import { Subscription, SubscriptionQuery } from "./types";
```
Comment (jeffda-self) [inline src/Account/Subscriptions/GetSubscriptions.ts:2]: Should use "import type".

File: src/Account/Subscriptions/UpdateSubscription.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import type { APICall, APIFetchSettings } from "../../types";
+import { Subscription, SubscriptionUpdatePaymentMethodBody, SubscriptionUpdateRenewalBody } from "./types";
```
Comment (jeffda-self) [inline src/Account/Subscriptions/UpdateSubscription.ts:2]: Should use "import type".

**Review by jeffda-self — CHANGES_REQUESTED** [review]

**Review by jeffda-self — APPROVED** [review]

**Review by jeffda-self — APPROVED** [review]

**Pull request merged**